src/restoreTerminals.ts

```diff
@@ -107,7 +131,31 @@ async function createNewSplitTerminal(
         break;
       }
       if (numTermsNow > numTermsBefore) {
-        resolve(vscode.window.terminals[numTermsNow - 1]);
+        const newTerm = vscode.window.terminals[numTermsNow - 1];
+
+        if (cwd) {
+          return vscode.workspace.fs.readDirectory(vscode.Uri.file(cwd)).then(
+            (data) => {
+              console.log({ data });
+              newTerm.sendText(`cd ${cwd}`, true);
+              return newTerm;
+            },
+            (err) => {
+              console.log({ err, cwd });
+
+              let errorMessage = err.message;
+
+              if (err?.code === "FileNotFound") {
+                errorMessage = `The terminal process failed to launch: Starting directory (cwd) "${cwd}" does not exist.`;
+              }
+
+              vscode.window.showErrorMessage(errorMessage);
+              newTerm.dispose();
+            }
+          );
+        }
+
+        resolve(newTerm);
```

Comment:
I think this resolve needs to be in the success callback?

I'd strongly prefer the `vscode.workspace.fs.readDirectory(vscode.Uri.file(cwd))` to be resolved with an `await` though (@EthanSK?)

Reply:
That `resolve` is for when you haven't specified `cwd`.

Here's what it would look like with an `await`:

```javascript
if (cwd) {
  try {
    await vscode.workspace.fs.readDirectory(vscode.Uri.file(cwd));
    newTerm.sendText(`cd ${cwd}`, true);
  } catch (err) {
    let errorMessage = err.message;

    if (err?.code === "FileNotFound") {
      errorMessage = `The terminal process failed to launch: Starting directory (cwd) "${cwd}" does not exist.`;
    }

    vscode.window.showErrorMessage(errorMessage);
    newTerm.dispose();
  }
}
```

I'm not sure which I prefer honestly.

Reply:
ACK